internal/linter/check_pk_type.go

```diff
@@ -0,0 +1,69 @@
+package linter
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/skeema/skeema/internal/tengo"
+)
+
+func init() {
+	rule := Rule{
+		CheckerFunc:     TableBinaryChecker(pkTypeChecker),
+		Name:            "pk-type",
+		Description:     "Only allow primary keys to have types listed in --allow-pk-type",
+		DefaultSeverity: SeverityIgnore,
+	}
+	rule.RelatedListOption(
+		"allow-pk-type",
+		"tinyint, smallint, mediumint, int, bigint, varbinary, binary, date, datetime, timestamp, time, year, decimal",
```

Comment:
I'm leaning more towards an empty-string default (requiring each company to configure the allow-list as needed), instead of a default specifically for memcomparable types:

* General aim is for Skeema to be compatible with other ecosystem tools but without introducing logic specific to one tool (e.g. there's nothing pt-osc or gh-ost specific)
* Some companies will want to restrict PK types for other reasons, not based on whether they're memory comparable. And for better or worse, varchar PKs are very very common.
* If keeping the default as memcomparable types, MariaDB's added types (uuid, inet4, inet6) should be here since these are binary representations. But that means this default value is getting quite long, and also means more future maintenance work any time MySQL or MariaDB introduce additional memcomparable data types
* Are there other types that are technically memcomparable? I'm wondering about spatial types, maybe also the bit type... and then there's weird cases like set and enum... that said, most of these weird edge cases are poor choices for PK anyway though
* Users may interpret this list as a set of recommended PK types, in which case mediumint feels kind of gross as a 3-byte-int abomination, but maybe that's just me :)

That all said, I could be convinced otherwise. In part it may depend how many other ecosystem tools have the same memcomparable PK restriction. That would also make the docs easier to write, since we could say the default value reflects what is needed for tools X and Y and Z.

Reply:
Let's go with empty. If our schema change becomes popular, you can always change the defaults in future.